web/src/frontend/classic/stores/Documentation.ts

```diff
@@ -0,0 +1,80 @@
+import { writable } from 'svelte/store';
+
+export type DocCategory = {
+    name: string;
+    description: HTMLElement;
+    sections: DocSection[];
+}
+export type DocSection = {
+    name: string,
+    content: HTMLElement
+}
+
+export const documentation = writable<DocCategory[]>([]);
+
+/**
+ * Creates a slide section by extracting HTML elements from the given content element until a specified split element is encountered.
+ * @param content - The content element from which to extract the slide section.
+ * @param split - The element name at which to stop extracting and split the slide section.
+ * @returns The created slide section element.
+ */
+function slideElement(content:HTMLElement, split:string[]) : HTMLElement {
+    const slide = document.createElement("div");
+    let nextSibling = content.nextSibling;
+    while (nextSibling && !split.includes(nextSibling.nodeName)) {
+        slide.appendChild(nextSibling.cloneNode(true));
+        nextSibling = nextSibling.nextSibling;
+    }
+    return slide;
+}
+
+(async function () {
+    const url = 'https://hakuneko.download/docs/haruneko/';
+    // TODO Documentation should be cached using a service worker
+    const cacheKey = `cache:${url}`;
+    const cacheTimestamp = localStorage.getItem(`${cacheKey}:timestamp`);
+    if (cacheTimestamp) {
+        const timestamp = parseInt(cacheTimestamp);
+        const oneWeek = 7 * 24 * 60 * 60 * 1000; // 1 week in milliseconds
+        const currentTime = Date.now();
+        if (currentTime - timestamp > oneWeek) {
+            localStorage.removeItem(cacheKey);
+            localStorage.removeItem(`${cacheKey}:timestamp`);
+        }
+    }
+    let content = localStorage.getItem(cacheKey);
+    if (!content?.length) {
+        const response = await fetch(url);
+        const text = await response.text();
+        localStorage.setItem(cacheKey, text);
+        localStorage.setItem(`${cacheKey}:timestamp`, Date.now().toString());
+        content = text;
+    }
+    const parser = new DOMParser();
+    const html = parser.parseFromString(content, 'text/html');
+    const main = html.querySelector('section');
+    const categories: DocCategory[] = [];
+    if (main) {
+        const images = main.querySelectorAll('img');
+        images.forEach((image) => {
+            const src = image.getAttribute('src');
+            if (src && !src.startsWith('http')) {
```

Comment:
This assumes `src` to be a relative URL, but it could be a different protocol or protocol-less in which case this transformation rule will fail.

Reply:
This is intentional as all img will be always  (at least more than probably) relative as per current documentation. Could be improved later.